### ISSUE TITLE
RSPEC-5757 Allowing confidential information to be logged is security-sensitive

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#1965]
Implements [RSPEC-5757](https://jira.sonarsource.com/browse/RSPEC-5757)

For both JS/TS

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5757

Report on `secrets: []` as main location and `new Signale` as secondary location:
```
const options = {
  secrets: []  // Sensitive main location
};
 
const logger = new Signale(options); // Sensitive secondary location
```

If `options` argument of `new Signale(options)` call doesn't contain `secret` property or `options` argument is not defined, the primary location should be on `new Signale()`:
```
const options = {
};
 
const logger1 = new Signale(options); // Sensitive
const logger2 = new Signale(); // Sensitive
```

**Comment by eric-therond-sonarsource** [comment]
Reopened because we have [1 FN on security-expected-issues](https://github.com/SonarSource/security-expected-issues/blob/master/javascript/rules/hotspots/RSPEC-5757/noncompliant2.js):
```
const logger = new Signale(); // FN
```
when no options/argument is passed to the Signale constructor, we should raise.
